homeassistant/components/light/hue.py

```diff
@@ -149,32 +150,53 @@ def update_lights():
             _LOGGER.exception("Cannot reach the bridge")
             return
 
-        api_states = api.get('lights')
+        api_lights = api.get('lights')
 
-        if not isinstance(api_states, dict):
+        if not isinstance(api_lights, dict):
+            _LOGGER.error("Got unexpected result from Hue API")
+            return
+
+        api_groups = api.get('groups')
+
+        if not isinstance(api_groups, dict):
             _LOGGER.error("Got unexpected result from Hue API")
             return
 
         new_lights = []
+        new_lightgroups = []
 
         api_name = api.get('config').get('name')
         if api_name in ('RaspBee-GW', 'deCONZ-GW'):
             bridge_type = 'deconz'
         else:
             bridge_type = 'hue'
 
-        for light_id, info in api_states.items():
+        for light_id, info in api_lights.items():
             if light_id not in lights:
                 lights[light_id] = HueLight(int(light_id), info,
                                             bridge, update_lights,
                                             bridge_type, allow_unreachable)
                 new_lights.append(lights[light_id])
             else:
                 lights[light_id].info = info
```

Comment:
Hmm, that would have to be done before the `new_lights.append(lights[light_id])` then, otherwise the object in `new_lights` won't have the `_state` variable. Then I also need this before the `lights[light_id].schedule_update_ha_state()` in the else-block, or it will be updated, but the `_state` isn't filled in yet giving the possibility of a race condition?

This will give this code:
```python
        for is_group, items in (False, api_lights.items()), (True, api_groups.items()):
            for light_id, info in items:
                if light_id not in lights:
                    lights[light_id] = HueLight(int(light_id), info, bridge,
                                                update_lights, bridge_type,
                                                allow_unreachable, is_group)
                    if is_group:
                        lights[light_id]._state = info['action']
                    else:
                        lights[light_id]._state = info['state']
                    new_lights.append(lights[light_id])
                else:
                    lights[light_id].info = info
                    if is_group:
                        lights[light_id]._state = info['action']
                    else:
                        lights[light_id]._state = info['state']
                    lights[light_id].schedule_update_ha_state()
```

I'm not sure if this will be cleaner then checking in `brightness`, `xy_color` and `color_temp`. I can't remove the `if is_group` in `is_on` since it reads a different variable (`any_on`).